index.js

```diff
@@ -169,4 +169,35 @@ const isEmptyMapOrSet = x => (is.map(x) || is.set(x)) && x.size === 0;
 
 is.empty = x => !x || isEmptyStringOrArray(x) || isEmptyObject(x) || isEmptyMapOrSet(x);
 
+const isAnyOrAll = (all, predicate, values) => {
```

Comment:
Instead of passing `all` as a boolean, pass in a `method` param instead.

this would actually be either `Array.prototype.some` or `Array.prototype.every`,

and you'd call `method.call(values, predicate)`.

it makes sense more than passing a boolean. If we're speaking clean code and `SOLID`, this violates both `Single Responsibility Principle` and `Open Close Principle`, because it makes `isAnyOrAll` (which should be renamed, for the same reasons) responsible for both having the the calls to the methods inside of it, and needing the logic to decide which method is called.

By passing the method from the caller, we can allow more methods to use `isAnyOrAll`.

I'm sure that there are more method of `Array` that receive a predicate.

Reply:
Makes sense! What would you suggest we change the name of `isAnyOrAll` to?

Reply:
predicateOnArray